### PR TITLE
make Travis-CI use default sbt launcher

### LINKED ISSUE #766: Move `travis.yml/default.yml` off sbt-extras, onto official launcher
As per discussion on https://github.com/scala/scala-dev/issues/765

Yes, there is a risk that somebody's repo will actually be expecting sbt-extras, but I think we should just rip that bandaid off, there is no "good time" to rip it off.

Motivation: sbt-extras is not the standard, and the proliferation of multiple launch scripts (there's a coursier one now, too) is bad for the community. I think we should discourage use or perception of sbt-extras as a de facto default. The official script should be the de facto default. It's fine for sbt-extras to exist as a power-user alternative for those who want it. (And if sbt-extras has some feature that is really so commonly needed, that feature should be ported to the official one.)

### PR DESCRIPTION
**Pull request by SethTisue**
fixes #766

### CODE REVIEW

File: travis/default.yml [hunk 1]
```diff
@@ -16,12 +16,11 @@ before_install:
...
   - java -version 2>&1 | grep 'OpenJDK.*[^0-9]'$versionPrefix'[^0-9]' || exit 1
-  # Travis-CI has (as of March 2021, anyway) an outdated sbt-extras version,
-  # so overwrite it with a March 2021 version that works with sbt 1.4.8+
+  # Travis-CI's default is an outdated version of sbt-extras;
+  # overwrite it with the official sbt launcher
   - |
-    curl -sL https://raw.githubusercontent.com/paulp/sbt-extras/dc4f350f112580fcdf5f6fa7e8d5d2116475f84a/sbt  > /tmp/sbt-launch-script || travis_terminate 1
-    chmod +x /tmp/sbt-launch-script || travis_terminate 1
-    sudo mv /tmp/sbt-launch-script /usr/local/bin/sbt || travis_terminate 1
+    curl -L --silent "https://raw.githubusercontent.com/sbt/sbt/1.5.x/sbt" > $HOME/sbt
```
Comment (lrytz) [inline travis/default.yml:22]: Maybe better stick to a specific tag or revision?

Comment (dwijnand) [reply travis/default.yml:22]: 1.5.x is the branch just for fixes, not for more experimental changes (develop).  So this LGTM

**Review by dwijnand — APPROVED** [review]

File: travis/default.yml (hunk 1 above)
Comment (lrytz) [reply travis/default.yml:22]: 👍 OK

**Comment by SethTisue** [comment]
after testing at https://github.com/scala/scala-parser-combinators/pull/413 this needed fixups: 7cc8821859e908ec46623b92e49f0066028a0df6, 6fdea9645b37fa8226b134a75ac80f15b3b8b008

**Pull request merged**